collector/agentPoller.go

```diff
@@ -15,451 +15,433 @@
+			Unit:      "",                                        // TODO(roger): not currently an easy way to get units
+			Value:     fmt.Sprintf("%v", v.Field(i).Interface()), // TODO(roger): everything is a string for MVP
+			Timestamp: t.UTC().Format(time.RFC3339Nano),
+		})
 	}
+	out = append(out, msg)
+
+	// Produce container metrics
+	for _, c := range in.containerMetrics {
+		msg = producers.MetricsMessage{
+			Name: strings.Join([]string{
+				producers.ContainerMetricPrefix,
+				c.ContainerID,
+			}, producers.MetricNamespaceSep),
+			Datapoints: []producers.Datapoint{},
+			Dimensions: producers.Dimensions{},
+			Timestamp:  t.UTC().Unix(),
+		}
+		v = reflect.Indirect(reflect.ValueOf(c.Statistics))
+		for i := 0; i < v.NumField(); i++ {
+			n := strings.Join([]string{
+				msg.Name,
+				strings.Split(v.Type().Field(i).Tag.Get("json"), ",")[0],
+			}, producers.MetricNamespaceSep)
+
+			msg.Datapoints = append(msg.Datapoints, producers.Datapoint{
+				Name:      strings.Replace(n, "/", producers.MetricNamespaceSep, -1),
+				Unit:      "",                                        // TODO(roger): not currently an easy way to get units
+				Value:     fmt.Sprintf("%v", v.Field(i).Interface()), // TODO(roger): everything is a string for MVP
+				Timestamp: t.UTC().Format(time.RFC3339Nano),
+			})
+		}
 
-	// state["id"] (agent_id)
-	agentID, err := json.GetString("id")
-	if err != nil {
-		stats <- statsd.MakeEvent(statsd.AgentQueryBadData)
-		return nil, err
-	}
+		fi, ok := getFrameworkInfoByFrameworkID(c.FrameworkID, in.agentState.Frameworks)
+		if !ok {
+			log.Warnf("Did not find FrameworkInfo for framework ID %s, skipping!", fi.ID)
+			continue
+		}
 
-	frameworks, err := json.GetObjectArray("frameworks")
-	if err != nil {
-		stats <- statsd.MakeEvent(statsd.AgentQueryBadData)
-		return nil, err
+		msg.Dimensions.AgentID = in.agentState.ID
+		msg.Dimensions.ClusterID = "" // TODO(roger) need to get this from the master
+		msg.Dimensions.ContainerID = c.ContainerID
+		msg.Dimensions.FrameworkID = c.FrameworkID
+		msg.Dimensions.FrameworkName = fi.Name
+		msg.Dimensions.FrameworkRole = fi.Role
+		msg.Dimensions.FrameworkPrincipal = fi.Principal
+		msg.Dimensions.ExecutorID = c.ExecutorID
+		msg.Dimensions.ContainerID = c.ContainerID
+		msg.Dimensions.Labels = getLabelsByContainerID(c.ContainerID, in.agentState.Frameworks)
+
+		out = append(out, msg)
 	}
 
-	// state["frameworks"][N]["id"] (framework_id)
-	// => state["frameworks"][N]["name"] (framework_name)
-	frameworkNames := make(map[string]string, len(frameworks))
-
-	// state["frameworks"][N]["executors"][M]["id"] (executor_id)
-	// => state["frameworks"][N]["executors"][M]["labels"][L(MARATHON_APP_ID)]["value"] (application_name)
-	executorAppNames := make(map[string]string, 0)
+	return out
+}
 
+// getFrameworkInfoByFrameworkID returns the FrameworkInfo struct given its ID.
+// For example: "5349f49b-68b3-4638-aab2-fc4ec845f993-0000" => "marathon"
```

Comment:
Nit: this example no longer applies; this function was generalized to return a `frameworkInfo` instead of a `frameworkInfo.Name`.